core/object/script_language.h

```diff
@@ -221,20 +221,75 @@ struct ScriptCodeCompletionOption {
 		KIND_FILE_PATH,
 		KIND_PLAIN_TEXT,
 	};
+
+	enum Location {
+		LOCATION_LOCAL = 0,
+		LOCATION_PARENT_MASK = (1 << 8),
+		LOCATION_OTHER_USER_CODE = (1 << 9),
+		LOCATION_OTHER = (1 << 10),
+	};
+
 	Kind kind = KIND_PLAIN_TEXT;
 	String display;
 	String insert_text;
 	Color font_color;
 	RES icon;
 	Variant default_value;
 	Vector<Pair<int, int>> matches;
+	int location = LOCATION_OTHER;
```

Comment:
Should be an enum.

Reply:
We use the `PARENT_MASK` enum value as a flag to indicate it comes from a parent - but then the other bits in the integer are used to store the depth of the parent, so it is not strictly an enum value.